api/src/main/java/io/strimzi/api/kafka/model/kafka/cruisecontrol/CruiseControlSpec.java

```diff
@@ -197,6 +201,19 @@ public void setTemplate(CruiseControlTemplate template) {
         this.template = template;
     }
 
+    @Description("Auto-rebalancing on scaling related configuration listing the modes, when brokers are added or removed, " +
+            "with the corresponding rebalance template configurations." +
+            "If this field is set, at least one mode has to be defined.")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @MinimumItems(1)
```

Comment:
Why? We do not seem to use it for other lists. I'm not sure I follow the value and it might just make it more complicated to modify the resource programmatically.

Reply:
The idea was to not allowing something like `autoRebalance: { }` which could confuse users to think that auto-rebalancing is configured with some default modes configurations which is not true. If you put the `autoRebalance` field, you need to specify at least one mode (then you can leave the template out for that mode and use default CC config). Do you see any better way to enforce it? Doing that in the operator code and not at API level?